src/core/drive/page_renderer.ts

```diff
@@ -104,9 +105,14 @@ export class PageRenderer extends Renderer<HTMLBodyElement, PageSnapshot> {
     }
   }
 
-  assignNewBody() {
+  async assignNewBody() {
+    await nextEventLoopTick()
```

Comment:
There are some more details on why this is necessary in https://github.com/hotwired/turbo/pull/305#issuecomment-1183505367, but in short, without this, snapshot caches don't happen at the right time, as [`cacheSnapshot`](https://github.com/hotwired/turbo/blob/98cdc4037788f078a741c7fb50412db0101bcb97/src/core/drive/page_view.ts#L40-L49) waits *just* long enough to end up caching the new content, rather than the original. While not an issue when replacing the entire `<body>` element (since that element has not changed), it is an issue when replacing content *within* the `<body>`.